src/Success.php

```diff
@@ -41,79 +42,41 @@ private function __construct($value)
     }
 
     /**
-     * Create a new error value.
+     * Create a new Success value.
      *
      * @template S
      *
      * @param S $value
      *
-     * @return \GrahamCampbell\ResultType\Result<S,E>
+     * @return \GrahamCampbell\ResultType\Result<S,never>
      */
     public static function create($value)
     {
         return new self($value);
     }
 
-    /**
-     * Get the success option value.
-     *
-     * @return \PhpOption\Option<T>
-     */
-    public function success()
+    public function success(): Option
     {
         return Some::create($this->value);
     }
 
-    /**
-     * Map over the success value.
-     *
-     * @template S
-     *
-     * @param callable(T):S $f
-     *
-     * @return \GrahamCampbell\ResultType\Result<S,E>
-     */
-    public function map(callable $f)
+    public function map(callable $f): Result
     {
         return self::create($f($this->value));
     }
 
-    /**
-     * Flat map over the success value.
-     *
-     * @template S
-     * @template F
-     *
-     * @param callable(T):\GrahamCampbell\ResultType\Result<S,F> $f
-     *
-     * @return \GrahamCampbell\ResultType\Result<S,F>
-     */
-    public function flatMap(callable $f)
+    public function flatMap(callable $f): Result
     {
         return $f($this->value);
     }
 
-    /**
-     * Get the error option value.
-     *
-     * @return \PhpOption\Option<E>
-     */
-    public function error()
+    public function error(): Option
     {
         return None::create();
     }
 
-    /**
-     * Map over the error value.
-     *
-     * @template F
-     *
-     * @param callable(E):F $f
-     *
-     * @return \GrahamCampbell\ResultType\Result<T,F>
-     */
-    public function mapError(callable $f)
+    public function mapError(callable $f): Result
     {
-        return self::create($this->value);
```

Comment:
i intentionally made a copy here

Reply:
@GrahamCampbell can we write a unit test somehow, which would fail if we don't create the copy? All existing ones pass, so it looks like redundant.

Reply:
The other reason not to bother with a copy is that Result is immutable, so calling methods on its reference is identical to calling its copy at any time.